crates/sound/src/audio_backends/pipewire.rs

```diff
@@ -407,29 +419,37 @@ impl AudioBackend for PwBackend {
 
     fn start(&self, stream_id: u32) -> Result<()> {
         debug!("pipewire start");
-        self.stream_params.write().unwrap()[stream_id as usize]
+        self.stream_params
+            .write()
+            .unwrap()
+            .get_mut(stream_id as usize)
+            .ok_or_else(|| Error::StreamWithIdNotFound(stream_id))?
             .state
             .start();
         let lock_guard = self.thread_loop.lock();
         let stream_hash = self.stream_hash.read().unwrap();
-        let Some(stream) = stream_hash.get(&stream_id) else {
-            return Err(Error::StreamWithIdNotFound(stream_id));
-        };
+        let stream = stream_hash
+            .get(&stream_id)
+            .expect("Could not find stream with this id in `stream_hash`.");
         stream.set_active(true).expect("could not start stream");
         lock_guard.unlock();
         Ok(())
     }
 
     fn stop(&self, stream_id: u32) -> Result<()> {
         debug!("pipewire stop");
-        self.stream_params.write().unwrap()[stream_id as usize]
+        self.stream_params
+            .write()
+            .unwrap()
+            .get_mut(stream_id as usize)
+            .ok_or_else(|| Error::StreamWithIdNotFound(stream_id))?
             .state
             .stop();
         let lock_guard = self.thread_loop.lock();
         let stream_hash = self.stream_hash.read().unwrap();
-        let Some(stream) = stream_hash.get(&stream_id) else {
-            return Err(Error::StreamWithIdNotFound(stream_id));
-        };
+        let stream = stream_hash
+            .get(&stream_id)
+            .expect("Could not find stream with this id in `stream_hash`.");
```

Comment:
IMHO a panic-based error handling approach could be risky and is not neccessary here, it's better to propagate the error like the previous code.

Reply:
The difference is that at this point we know `stream_id` is valid (else we would have returned the error earlier). So if there's no `stream_id` in `stream_hash` the state is invalid, hence the panic.

Reply:
ok I see make sense.